### PR TITLE
refactor(sdk-coin-bsc): introduce eth like coin to inherit from

### PR DESCRIPTION
**Pull request by kxl4126**
Ticket: BG-56005

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/abstract-eth/src/EthLikeToken.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    const chain = coinNames.Mainnet;
+    for (const token of [...tokens.bitcoin[chain].tokens, ...tokens.testnet[chain].tokens]) {
+      const tokenConstructor = this.createTokenConstructor(token, coinNames);
+      tokensCtors.push({ name: token.type, coinConstructor: tokenConstructor });
+      tokensCtors.push({ name: token.tokenContractAddress, coinConstructor: tokenConstructor });
+    }
+    return tokensCtors;
+  }
+
+  // static coinNames() {
```
Comment (wesgraham) [inline modules/abstract-eth/src/EthLikeToken.ts:38]: nit: remove?

File: modules/abstract-eth/src/EthLikeToken.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    this.tokenConfig = tokenConfig;
+  }
+
+  static createTokenConstructor(config: EthLikeTokenConfig, coinNames: CoinNames): CoinConstructor {
+    return (bitgo: BitGoBase) => new this(bitgo, config, coinNames);
+  }
+
+  static createTokenConstructors(coinNames: CoinNames): NamedCoinConstructor[] {
+    const tokensCtors: NamedCoinConstructor[] = [];
+    const chain = coinNames.Mainnet;
```
Comment (wesgraham) [inline modules/abstract-eth/src/EthLikeToken.ts:29]: would this work? or are tokens.testnet indexed by mainnet token names?

**Review by wesgraham — DISMISSED** [review]

**Review by wesgraham — COMMENTED** [review]
approved once nits are addressed

File: modules/abstract-eth/src/EthLikeToken.ts (hunk 1 above)
Comment (kxl4126) [reply modules/abstract-eth/src/EthLikeToken.ts:38]: Oh yep, will do

File: modules/abstract-eth/src/EthLikeToken.ts (hunk 2 above)
Comment (kxl4126) [reply modules/abstract-eth/src/EthLikeToken.ts:29]: So the code before just hardcoded `chain` to be `bsc` (in the case of binance tokens), so I assumed that we always use the mainnet chain name. It seems that the testnet tokens are in `...tokens.testnet[chain].tokens]` and mainnet tokens are in `...tokens.bitcoin[chain].tokens]`

**Review by wesgraham — DISMISSED** [review]

**Review by byhow — DISMISSED** [review]

File: modules/abstract-eth/src/index.ts [hunk 3]
```diff
@@ -1 +1,2 @@
 export * from './abstractEthLikeCoin';
+export * from './EthLikeToken';
```
Comment (mmcshinsky-bitgo) [inline modules/abstract-eth/src/index.ts:2]: nit: match file casing

**Review by byhow — APPROVED** [review]

**Pull request merged**